### PR TITLE
Replace plan/join.rs unit tests to use AST builder!

### PR DESCRIPTION
**Pull request by panarch**

### CODE REVIEW

**Review by devgony — APPROVED** [review]
It looks so neat and has way higher readability!
Thank you for all the efforts!

File: core/src/plan/join.rs [hunk 1]
```diff
@@ -765,31 +596,13 @@ mod tests {
...
-                group_by: Vec::new(),
-                having: None,
-            })
-        };
-        assert_eq!(actual, expected, "complex where_clause:\n{sql}");
+        let expected = table("Player")
+            .select()
+            .left_join("PlayerItem")
+            .hash_executor("PlayerItem.user_id", "Player.id")
+            .hash_filter("PlayerItem.amount > 10 AND PlayerItem.amount * 3 <= 2")
```
Comment (ever0de) [inline core/src/plan/join.rs:603]: (Hope) It would be nice to have `AND` as a chaining method.

File: core/src/plan/join.rs [hunk 2]
```diff
@@ -898,69 +663,20 @@ mod tests {
...
-                    left: Box::new(expr("u1.id")),
-                    op: BinaryOperator::Eq,
-                    right: Box::new(Expr::Subquery(Box::new(subquery))),
-                }),
-                group_by: Vec::new(),
-                having: None,
-            })
-        };
-        assert_eq!(actual, expected, "hash join in nested subquery:\n{sql}");
+        let expected = table("Player").alias_as("u1").select().filter(
```
Comment (zmrdltl) [inline core/src/plan/join.rs:666]: I am so impressed by the increase in readability through the query builder function that has such a long ast.

**Review by ever0de — APPROVED** [review]
I think I saw a piece of history

**Pull request merged**